### PR TITLE
interop: Extend passive client interop tests.

### PR DESCRIPTION
**Pull request by duesee**

### CODE REVIEW

File: openmls/src/schedule/psk.rs
Comment (franziskuskiefer) [inline openmls/src/schedule/psk.rs:350]: -19?

File: Cargo.toml
Comment (franziskuskiefer) [inline Cargo.toml:19]: A reminder to change this before merging.

File: openmls/src/group/public_group/validation.rs
Comment (franziskuskiefer) [inline openmls/src/group/public_group/validation.rs:328]: Do we really need to clone here? We shouldn't need to take any values here.

File: openmls/src/group/public_group/validation.rs
Comment (franziskuskiefer) [inline openmls/src/group/public_group/validation.rs:314]: This doesn't look right. Only resumption and external are PSK types. Application is a PSK usage. It looks like you're doing the right thing. But the documentation needs to updated to make clear what's what.

File: openmls/src/group/public_group/validation.rs
Comment (franziskuskiefer) [inline openmls/src/group/public_group/validation.rs:365]: Ah so you're storing the entire PskId. Wouldn't a hash or something be enough?
Anyway, this is fine for now.

File: openmls/src/group/public_group/validation.rs
Comment (franziskuskiefer) [inline openmls/src/group/public_group/validation.rs:364]: `contains` can be expensive if there are many PSK proposals. This probably won't be the case most of the time. But if you want to keep using a vector, please file an issue and link it here for potential performance bottlenecks/improvements.

File: openmls/src/group/public_group/validation.rs
Comment (franziskuskiefer) [inline openmls/src/group/public_group/validation.rs:373]: Remove?

File: openmls/src/schedule/psk.rs
Comment (franziskuskiefer) [inline openmls/src/schedule/psk.rs:135]: While you're here, can you add some short docs and remove this?

File: openmls/src/group/core_group/kat_passive_client.rs
Comment (franziskuskiefer) [inline openmls/src/group/core_group/kat_passive_client.rs:183]: Can you use the GH syntax for all of these (#1330) to get better usability?

**Comment by franziskuskiefer** [comment]
Oh, and we need some tests for the new valsems.

File: openmls/src/group/public_group/validation.rs
Comment (duesee) [reply openmls/src/group/public_group/validation.rs:314]: I used `External` and `Resumption/<usage>` as shorthands but this is not very clear. Will update :+1:

File: openmls/src/group/public_group/validation.rs
Comment (duesee) [reply openmls/src/group/public_group/validation.rs:373]: Oh, no, sorry! This can only be removed after #1330 was closed. We use it in testing for now to skip PreSharedKey proposals.

File: openmls/src/group/public_group/validation.rs
Comment (duesee) [reply openmls/src/group/public_group/validation.rs:364]: I'll use a `BTreeSet`.

File: Cargo.toml
Comment (duesee) [reply Cargo.toml]: I think it's not needed to block on this. I reverted to `0.0.3-pre.1` and opened #1334.

File: openmls/src/group/core_group/kat_passive_client.rs
Comment (franziskuskiefer) [inline openmls/src/group/core_group/kat_passive_client.rs:139]:
```suggestion
            cipher_suite,
```

File: openmls/src/group/public_group/validation.rs
Comment (franziskuskiefer) [inline openmls/src/group/public_group/validation.rs:324]: If they are always filtered, we can skip this here.
I guess it depends on the definition of "duplicate". I'm fine with keeping this as is but maybe make a note in an issue (e.g. validation) to double check that we don't do unnecessary work.

File: openmls/src/group/tests/test_commit_validation.rs
Comment (franziskuskiefer) [inline openmls/src/group/tests/test_commit_validation.rs:268]: Can you add to the comment why this isn't working anymore and what's happening in #1330 to change this? Same below.

File: openmls/src/group/tests/test_proposal_validation.rs
Comment (franziskuskiefer) [inline openmls/src/group/tests/test_proposal_validation.rs:1977]: These commented out tests should also get the issue referenced that will look into whether we need to do this or not.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, only a few nits then it can go in.

* Can you add something to the open interop issues for what's still open here?

**Review by franziskuskiefer — APPROVED** [review]
Thanks

**Pull request merged**
